### PR TITLE
Select hovered suggestion in strategy-explorer

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: runtime/browser/demo/explorer.html [hunk 1]
```diff
@@ -27,6 +27,15 @@
         e.preventDefault();
         explorerVisible = !explorerVisible;
         explorer.style.display = explorerVisible ? 'block' : '';
+        let selectedHash = document.body.querySelector('demo-flow').shadowRoot.querySelector('suggestions-element')._selectedHash;
```
Comment (mmandlis) [inline runtime/browser/demo/explorer.html]: this part is really ugly - what's a better way to do it?

Comment (sjmiles) [reply runtime/browser/demo/explorer.html]: Good catch, it's important that these objects not be tightly coupled to each other. Even when these modules are not strictly Web Components, we should use WC design concepts for encapsulation and reusability. The short answer is to handle this similarly to `generations` (which is to say, via *events*). The long answer has to to with application of Mediator Pattern: some Mediator owns objects and mediates their communication, allowing the objects themselves to be ignorant of their environment and enabling reuse.

If that's not enough of a hint, or if there is a wrinkle in this particular problem, please follow up.

Comment (sjmiles) [reply runtime/browser/demo/explorer.html]: Also, almost any use of `.shadowRoot` is breaking encapsulation and should be avoided (it usually means the element that owns the root needs more API to allow public access to the needed info).

Comment (mmandlis) [reply runtime/browser/demo/explorer.html]: I updated the code, could you, please, take another look? Thanks.

**Pull request merged**